spotguide/merge.go

```diff
@@ -0,0 +1,146 @@
+// Copyright © 2018 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spotguide
+
+import "reflect"
+
+func maxLength(x, y int) int {
+	if x < y {
+		return y
+	}
+	return x
+}
+
+func elementOrNil(arr []interface{}, i int) interface{} {
+	if i < len(arr) {
+		return arr[i]
+	}
+
+	return nil
+}
+
+func getKeys(obj map[string]interface{}) []string {
+	keys := reflect.ValueOf(obj).MapKeys()
+	strkeys := make([]string, len(keys))
+	for i := 0; i < len(keys); i++ {
+		strkeys[i] = keys[i].String()
+	}
+
+	return strkeys
+}
+
+func mergeKeys(a, b []string) []string {
+	keys := append(a, b...)
+
+	seen := make(map[string]bool)
+	uniqKeys := []string{}
+	for _, entry := range keys {
+		if _, value := seen[entry]; !value {
+			seen[entry] = true
+			uniqKeys = append(uniqKeys, entry)
+		}
+	}
+
+	return uniqKeys
+}
+
+func merge(dst, src interface{}) (interface{}, error) {
+	if reflect.DeepEqual(dst, src) {
+		return dst, nil
+	}
+
+	switch dstV := dst.(type) {
+	// string
```

Comment:
do we really need these comments?

Reply:
I prefer to leave them, to make sure all the JSON types are handled